### ISSUE TITLE
Profiling keras app with timeline, found the first dim of tensor is different with the batch size

### ISSUE THREAD

**Issue by seasonyc** [keras-team/keras#11374]
I profiled my network with tensorflow timeline, and checked the shape of the tensor, it looked strange, the first dim is not same as but less than the batch size. For different network, the first dim may be different, but one network has its max value, the first dim is always the "max value" if the batch size is more than the "max value", and the first dim looks to be a number of 2^n most down-close to the batch size if the batch size is less than the "max value". I tested on another CPU version, the first dim is same as the GPU's

What confused me is: I accumulated the allocated_bytes of the tensors, the number is obviously different with the real usage, in my test code, I checked the memory usage by tf.contrib.memory_stats.MaxBytesInUse(). When I changed the batch size, the number of tf.contrib.memory_stats.MaxBytesInUse() was changed with it, and allocated_bytes of the tensors always had not any change...

I test another network with writing by tensorflow raw api, the first dim of tensor is always same as the batch size. The accumulated allocated_bytes of the tensors are almost same as tf.contrib.memory_stats.MaxBytesInUse(). When I changed the batch size, the allocated_bytes of the tensors were certainly changed with it.

So I think there should be some issue.

I tested on 2 different PCs:
One is Windows 7, tensorflow-gpu 1.11, keras 2.2.4, cuda 9, cudnn9v7, GTX 1070
Another is Windows 7, tensorflow 1.11, keras 2.2.4

You can clone https://github.com/seasonyc/timeline_test, and then run timeline_test_keras.py and timeline_test_tf2.py(which is actually based on a tensorflow example code) to generate the timeline trace and print the value of MaxBytesInUse() and the accumulated allocated_bytes according to the timeline trace.

Thanks
YC

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
